pulp_smash/resources/platform/api_v2/api.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+"""Api package contains classes for easier resource managament in pulp-smash.
+"""
+
+from __future__ import unicode_literals
+
+import requests
+import time
+from requests.exceptions import HTTPError
+from pulp_smash.config import get_config
+
+# List of all paths here
+CREATE_REPOSITORY_PATH = "/pulp/api/v2/repositories/"
+REPOSITORY_PATH = "/pulp/api/v2/repositories/{}/"  # .format(<repo_id>)
+REPOSITORY_PATH = "/pulp/api/v2/repositories/{}/"  # .format(<repo_id>)
+POLL_TASK_PATH = "/pulp/api/v2/tasks/{}/"  # .format(<task_id>)
+
+#  Repository related variables
+REPORT_KEYS = {
+    'result',
+    'error',
+    'spawned_tasks',
+}
+ERROR_KEYS = {
+    '_href',
+    'error',
+    'error_message',
+    'exception',
+    'http_status',
+    'traceback',
+}
+
+# Task related variables
+TASK_ERROR_STATES = {
+    'error',
+    'timed out',
+}
+TASK_FINISHED_STATES = {
+    'finished',
+}
+
+
+class Repository(object):
+    """Provides interface for easy manipulation with pulp repositories.
+    `Create repo` accepts following kwarg parameters:
+        .. _Create repo:
+            http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/cud.html
+
+    Each time request to server is made, ie. by calling :meth:`create_repo`
+    method, response is saved to last_response variable.
+
+    :param id: System wide unique repository identifier.
+    :param display_name: User-friendly name for the repository
+    :param description: User-friendly text describing the repository’s contents
+    :param notes: Key-value pairs to programmatically tag the repository
+    :param importer_type_id: Type id of importer being associated with the
+        repository
+    :param importer_config: Configuration the repository will use to drive
+        the behavior of the importer
+    :distributors: Array of objects containing values of distributor_type_id,
+        repo_plugin_config, auto_publish, and distributor_id
+        """
+
+    def __init__(self, **kwargs):
+        self.data_keys = kwargs
+        self.last_response = None
+        self.cfg = get_config()
+
+    def create_repo(self, **kwargs):
+        """Create repository on pulp server.
+        After calling this method, <repo>.last_response.raise_for_status()
+        should be called in order to make sure that repo was correctly created.
+        :param kwargs: Additional arguments which will be passed to request,
+        same as in :class:`Repository` constructor.
+        """
+        self.data_keys.update(kwargs)
+        self.last_response = requests.post(
+            self.cfg.base_url + CREATE_REPOSITORY_PATH,
+            json=self.data_keys,
+            **self.cfg.get_requests_kwargs()
+        )
+
+    def delete_repo(self):
+        """Delete repository from pulp server.
+        After calling this method, <repo>.last_response.raise_for_status()
```

Comment:
This is syntactically valid, but its semantics are probably not what you intend. reStructuredText denotes a new paragraph with a blank line, and there is no blank line here, so this is interpreted as one giant paragraph. See [_A reStructuredText Primer_](http://docutils.sourceforge.net/docs/user/rst/quickstart.html).